### PR TITLE
network: rename SocketAddressProvider as ConnectionInfoProvider

### PR DESCRIPTION
**Pull request by soulxu**
Commit Message: network: rename SocketAddressProvider as ConnectionInfoProvider
Additional Description:
Since SocketAddressProvider isn't only include the socket address info, also include the information for connection, like connection id, SSL connection info and SNI name. So rename it as ConnectionInfoProvider.
Risk Level: low
Testing: all
Docs Changes: n/a
Release Notes: n/a
Part of #17168

### CODE REVIEW

**Comment by dmitri-d** [comment]
I realise the changes in the PR are renames (only?), but I think this is too large to review. Perhaps consider renaming the classes but leave the variable names as they are? Those can be renamed in smaller chunks, or as drive-bys? 

Should `ConnectionInfoProvider` and `ConnectionInfoProviderSetter` be moved into a dedicated header file?

**Comment by soulxu** [comment]
> I realise the changes in the PR are renames (only?), but I think this is too large to review. Perhaps consider renaming the classes but leave the variable names as they are? Those can be renamed in smaller chunks, or as drive-bys?
> 

yes, it is only about rename. also agree it is huge...terrible for review. I can separate them. Thanks for review this terrible PR!

> Should `ConnectionInfoProvider` and `ConnectionInfoProviderSetter` be moved into a dedicated header file?

Yes, it could be, do you want me do it in separate PR?

**Comment by soulxu** [comment]
It is still huge, I'm trying to only rename the classname without the member name first.

**Comment by dmitri-d** [comment]
> Yes, it could be, do you want me do it in separate PR?

Don't think it would increase the number of changes *that* much, I think it's probably ok to do it here. It's up to you, I think either way is ok though.

**Comment by dmitri-d** [comment]
>  I think either way is ok though.

Perhaps a dedicated PR would be better after all. I realised a bunch BUILD files woould need to be updated. that would make this PR harder to review...

**Comment by dmitri-d** [comment]
lgtm, thanks!

File: envoy/network/connection.h [hunk 1]
```diff
@@ -194,8 +194,8 @@ class Connection : public Event::DeferredDeletable,
   /**
    * @return the address provider backing this connection.
    */
-  virtual const SocketAddressProvider& addressProvider() const PURE;
-  virtual SocketAddressProviderSharedPtr addressProviderSharedPtr() const PURE;
+  virtual const ConnectionInfoProvider& addressProvider() const PURE;
```
Comment (mattklein123) [inline envoy/network/connection.h:197]: Your plan is to rename the function name and other things in follow ups, right?

/wait-any

Comment (soulxu) [reply envoy/network/connection.h:197]: yes, if we do that in one pr, it will be thousand lines.

Comment (mattklein123) [reply envoy/network/connection.h:197]: OK sounds good, thanks.

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**